### PR TITLE
Fix performance regression caused by #5748

### LINKED ISSUE #6137: Emerging seems slower. Lots of CPU spent in getSmoothLighting.
Unfortunately I will likely not have much time to track this down.

I observed that in recent MT builds the emerging seems much slower. In the profiler a lot of time is spent now in `getSmoothLight`. I do not recall ever seeing this is earlier profiler sessions.

git history `mapblock_mesh.cpp` show quite a bunch of recent changes.

### PR DESCRIPTION
**Pull request by numberZero**
should fix #6137

### CODE REVIEW

**Comment by numberZero** [comment]
@lhofhansl

**Comment by lhofhansl** [comment]
Doesn't compile :) (default: needs some statement)
When I fix it - removing the default: - I can confirm that the performance problem is fixed.

File: src/content_mapblock.cpp [hunk 1]
```diff
@@ -1288,14 +1288,20 @@ void MapblockMeshGenerator::errorUnknownDrawtype()
 
 void MapblockMeshGenerator::drawNode()
 {
+	// skip some drawtypes early
+	switch (f->drawtype) {
+		case NDT_NORMAL:   // Drawn by MapBlockMesh
+		case NDT_AIRLIKE:  // Not drawn at all
+		case NDT_LIQUID:   // Drawn by MapBlockMesh
+			return;
+		default:
```
Comment (bigfoot547) [inline src/content_mapblock.cpp]: You must place a statement inside the `default` label.

Comment (nerzhul) [inline src/content_mapblock.cpp]: yeah a break is required

File: src/content_mapblock.cpp [hunk 2]
```diff
@@ -1288,14 +1288,20 @@ void MapblockMeshGenerator::errorUnknownDrawtype()
...
+	}
+	origin = intToFloat(p, BS);
```
Comment (nerzhul) [inline src/content_mapblock.cpp]: sounds wrong to unroll this from loop no ? as p is moving in the loop

Comment (lhofhansl) [inline src/content_mapblock.cpp]: Yeah, `p` is now referring to the member `p`. That seems wrong.

Comment (lhofhansl) [inline src/content_mapblock.cpp]: Actually, this referred to the member `p` before as well... So on second thought it looks correct.

Comment (SmallJoker) [inline src/content_mapblock.cpp]: This is entirely correct. Now, the origin is only modified (calculated) when there's a node that actually needs to be drawn. Not much of an optimization but might also help a little. ;)

**Comment by paramat** [comment]
Tested with a break added and works, does seem faster :+1:

**Comment by paramat** [comment]
Edited and merged 20936e1c3e3ccc0bea2fbf212ebfff11f3cda8c3

**Pull request closed without merging**